docs/user_guide.md

```diff
@@ -719,6 +721,200 @@ If you are a project admin, you can disable a Robot Account by clicking "Disable
 If you are a project admin, you can delete a Robot Account by clicking "Delete" in the `Robot Accounts` tab of a project.
 ![delete_robot_account](img/robotaccount/disable_delete_robot_account.png)
 
+## Tag Retention Rules
+
+A repository can rapidly accumulate a large number of image tags, many of which might not be required after a given time or once they have been superseded by a subsequent image build. These excess tags can obviously consume large quantities of storage capacity. As a system administrator, you can define rules that govern how many tags of a given repository to retain, or for how long to retain certain tags. 
+
+### How Tag Retention Rules Work
+
+You define tag retention rules on repositories, not on projects. This allows for greater granularity when defining your retention rules. As the name suggests, when you define a retention rule for a repository, you are identifying which tags to retain. You do not define rules to explicitly remove tags. Rather, when you set a rule, any tags in a repository that are not identified as being eligible for retention are discarded. 
+
+A tag retention rule has 3 filters that are applied sequentially, as described in the following table.
+
+|Order|Filter|Description|
+|---|---|---|
+|First|Repository or repositories|Identify the repository or repositories on which to apply the rule. You can identify repositories that either have a certain name or name fragment, or that do not have that name or name fragment. Wild cards (for example `*repo`, `repo*`, and `**`) are permitted.|

```

Comment:
@stuclem Hi Stuart, this statement is correct but I want to point something out about repository matching since it behaves differently from tag matching.  The matched repos are the ones that the retention policies are actually performed on, the non-matched repos are not deleted/discarded, they are simply ignored (from this rule).  In other words, there is an ordering to this matching; repo matching happens first and these are 'earmarked' for further matching based on the tag criteria.  This was a design choice to minimize the area of images to be delete.  Not sure if it's worth pointing this out in the user_guide.

Reply:
I think that this info is useful. Added the following in https://github.com/goharbor/harbor/pull/8869/commits/bb32b5dbb292ef429652f14bca203d19b4d282bd:

The repository filter is applied first to mark the repositories to which to apply the retention rule. The identified repositories are earmarked for further matching based on the tag criteria. No action is taken on the nonspecified repositories at this stage.

Reply:
lgtm